### PR TITLE
utils: Convert Windows args to utf-8 string

### PR DESCRIPTION
**Pull request by ken2812221**
Create a new class `WinCmdLineArgs` when building for Windows. It converts all command line arguments to utf8 string.

### CODE REVIEW

**Comment by donaloconnor** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/13883/commits/f059e7a6269afd8449822c65ce1208cbd703895f

**Comment by alexeyneu** [comment]
no-bowler   
https://imageshack.com/a/img922/8836/nesafY.jpg  
```c++  
	std::wstring h;
	h=GetCommandLineW();

	std::wcout << h;
```
it's for wwinmain

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <util.h>
 #include <utilstrencodings.h>
 
+#include <tuple>
```
Comment (practicalswift) [inline src/bitcoin-cli.cpp]:
```
src/bitcoin-cli.cpp:18:1: warning: #includes are not sorted properly [llvm-include-order]
```

**Comment by MarcoFalke** [comment]
1e23b1cba1645b50570e1b4ca4f2da03e000f732 is adding a binary file. Is that wanted?

**Comment by ken2812221** [comment]
@MarcoFalke Sorry, forgot to delete that

File: src/bitcoin-cli.cpp [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #include <utilstrencodings.h>
 
 #include <memory>
+#include <tuple>
```
Comment (practicalswift) [inline src/bitcoin-cli.cpp]: Minor nit:

```
2018-09-13 21:33:38 clang-tidy(pr=13883): src/bitcoin-cli.cpp:18:1: warning: #includes are not sorted properly [llvm-include-order]
```

Comment (ken2812221) [reply src/bitcoin-cli.cpp]: Would you prefer to put tuple after stdio.h?

Comment (practicalswift) [reply src/bitcoin-cli.cpp]: I don't think we separate C and C++ headers elsewhere, but please sort them.

File: src/util.cpp [hunk 3]
```diff
@@ -1256,3 +1262,29 @@ int ScheduleBatchPriority(void)
...
+{
+    delete[] argv;
+}
+
+std::pair<int, char**> WinCmdLineArgs::get()
+{
+    return std::make_pair(argc, argv);
+}
+#endif
+}
```
Comment (practicalswift) [inline src/util.cpp]:
```
2018-09-13 21:33:38 clang-tidy(pr=13883): src/util.cpp:1294:2: warning: namespace 'util' not terminated with a closing comment [google-readability-namespace-comments]
```

**Comment by NicolasDorier** [comment]
Can you do a unit test on this?

**Comment by MarcoFalke** [comment]
Concept ACK 380c843217139b180457889699c65b37ae3b4a87 (Only checked that this should only affect Windows. Didn't review nor checked that the tests fail without this fix on windows.)

**Comment by NicolasDorier** [comment]
tACK

**Pull request merged**